compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/DerivedConfigUtils.cpp

```diff
@@ -11,28 +11,41 @@
 #include "iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.h"
 #include "mlir/Dialect/Linalg/IR/LinalgInterfaces.h"
 #include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypeInterfaces.h"
 #include "mlir/IR/TypeUtilities.h"
 
 namespace mlir::iree_compiler::IREE::GPU {
 
 static constexpr int64_t kPreferredCopyNumBits = 128;
 
-SmallVector<int64_t>
+// Helper to construct a list of tile sizes that simply uses the given vector
+// size or the innerDimSize as the inner most tile size, whichever is smaller.
+// All other dims are tiled to 1.
```

Comment:
I think this is a bit conservative/missing some special cases. Maybe add a TODO to improve the tile size selection here?

Reply:
This is actually what we want probably in the static nicely aligned case also. This will translate to all of the threads naturally doing as large of a contiguous read as possible in a for loop. The trick is then to unroll or fold away that for loop, but since those patterns aren't fully fleshed out, I left the existing behavior as is.